src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineTransport.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\Serialization\Serializer;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class DoctrineTransport implements TransportInterface
+{
+    private $connection;
+    private $serializer;
+    private $receiver;
+    private $sender;
+
+    public function __construct(Connection $connection, Serializer $serializer = null)
+    {
+        $this->connection = $connection;
+        $this->serializer = $serializer ?? Serializer::create();
+    }
+
+    public function receive(callable $handler): void
+    {
+        ($this->receiver ?? $this->getReceiver())->receive($handler);
```

Comment:
Never saw `??` used like this before. It's a bit puzzling at first, but it's a nice trick :slightly_smiling_face:

Reply:
Me neither. I took it from the Amqp Transport 😉

Reply:
IMHO `?:` seems to fit better as `$this->receiver` is always a defined var (although with `null` value), but I guess it doesn't matter and the result is the same at the end.